server/src/infra/instrumentation.ts

```diff
@@ -0,0 +1,45 @@
+import { serverVersion } from '@app/domain/domain.constant';
+import { AsyncLocalStorageContextManager } from '@opentelemetry/context-async-hooks';
+import { PrometheusExporter } from '@opentelemetry/exporter-prometheus';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { IORedisInstrumentation } from '@opentelemetry/instrumentation-ioredis';
+import { NestInstrumentation } from '@opentelemetry/instrumentation-nestjs-core';
+import { PgInstrumentation } from '@opentelemetry/instrumentation-pg';
+import { Resource } from '@opentelemetry/resources';
+import { ExplicitBucketHistogramAggregation, View } from '@opentelemetry/sdk-metrics';
+import { NodeSDK } from '@opentelemetry/sdk-node';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { OpenTelemetryModuleOptions } from 'nestjs-otel/lib/interfaces';
+
+const aggregation = new ExplicitBucketHistogramAggregation(
+  [0.1, 0.25, 0.5, 1, 2.5, 5, 10, 25, 50, 100, 250, 500, 1000, 2500, 5000],
+  true,
+);
+
+const otelSDK = new NodeSDK({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: `immich`,
+    [SemanticResourceAttributes.SERVICE_VERSION]: serverVersion.toString(),
+  }),
+  metricReader: new PrometheusExporter({ port: 8081 }),
+  contextManager: new AsyncLocalStorageContextManager(),
+  instrumentations: [
+    new HttpInstrumentation(),
+    new IORedisInstrumentation(),
+    new NestInstrumentation(),
+    new PgInstrumentation(),
+  ],
+  views: [new View({ aggregation, instrumentName: '*', instrumentUnit: 'ms' })],
+});
+
+export const otelConfig: OpenTelemetryModuleOptions = {
+  metrics: {
+    hostMetrics: process.env.OTEL_SDK_DISABLED !== 'true',
+    apiMetrics: {
+      enable: process.env.OTEL_SDK_DISABLED !== 'true',
+      ignoreRoutes: ['/favicon.ico', '/custom.css'],
```

Comment:
Why are these routes ignored? I think there is a list somewhere else in the app, should this be kept in sync with those routes?

Reply:
They add noise to the API metrics and it isn't really worth storing metric data for them.

Reply:
I can't find the list you mentioned. Do you know where I can find it?

Reply:
I moved the `excludePaths` in `main.ts` to `infra.config.ts` and have it shared here. I added `favicon.ico` to that, but not sure if that's okay